src/en/randowiz/src/eu/kanade/tachiyomi/extension/en/randowiz/Randowiz.kt

```diff
@@ -0,0 +1,154 @@
+        return Observable.just(
+            MangasPage(
+                listOf(
+                    SManga.create().apply {
+                        title = "Randowiz: We live in an MMO!?"
+                        artist = "Randowiz"
+                        author = "Randowiz"
+                        status = SManga.ONGOING
+                        url = "/category/we-live-in-an-mmo/"
+                        description =
+                            "The world of 'Mamuon' where players and NPC's live together in harmony. Or do they? DO THEY?"
+                        thumbnail_url =
+                            "https://i0.wp.com/randowis.com/wp-content/uploads/2016/02/MMO_CHP_001_CSP_000.jpg?resize=800%2C800&ssl=1"
+                    },
+                    SManga.create().apply {
+                        title = "Randowiz: Short comics"
+                        artist = "Randowiz"
+                        author = "Randowiz"
+                        status = SManga.ONGOING
+                        url = "/category/short-comics/"
+                        description =
+                            "So short that i have to compensate.."
+                        thumbnail_url =
+                            "https://i0.wp.com/randowis.com/wp-content/uploads/2021/10/Images_PNGs_Site_BOT-SUPPORT.png"
+                    },
+                    SManga.create().apply {
+                        title = "Randowiz: Illustations"
+                        artist = "Randowiz"
+                        author = "Randowiz"
+                        status = SManga.ONGOING
+                        url = "/category/art/"
+                        description =
+                            "You like draw? I give you draw."
+                        thumbnail_url =
+                            "https://i0.wp.com/randowis.com/wp-content/uploads/2021/05/colour-studies-021-post.jpg"
+                    }
+                ),
+                false
+            )
+        )
+    }
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList
+    ): Observable<MangasPage> = Observable.just(MangasPage(emptyList(), false))
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> = Observable.just(manga)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val chapters = mutableListOf<SChapter>()
+        chapters.addAll(document.select(chapterListSelector()).map { chapterFromElement(it) })
+        val next = document.select(".next").attr("href") ?: ""
+
+        if (next.isNotEmpty()) client.newCall(GET(next))
+            .asObservableSuccess()
+            .map { newWebpage ->
+                chapterListParse(newWebpage)
+            }.forEach { a -> chapters.addAll(a) }
+        return chapters.mapIndexed {
+            i, ch ->
+            ch.apply { chapter_number = chapters.size.toFloat() - i }
+        }
+    }
```

Comment:
You don't need to deal with observables and neither recursivity if you do like this:

```suggestion
    override fun chapterListParse(response: Response): List<SChapter> {
        val document = response.asJsoup()
        val chapters = document.select(chapterListSelector())
            .map { chapterFromElement(it) }
            .toMutableList()
        var next = document.selectFirst(".next")?.attr("href") ?: ""
        
        while (next.isNotEmpty()) {
            val nextDocument = client.newCall(GET(next, headers)).execute().asJsoup()
            
            chapters += nextDocument.select(chapterListSelector())
                .map{ chapterFromElement(it) }
            next = nextDocument.selectFirst(".next")?.attr("href") ?: ""
        }
        
        return chapters.mapIndexed { i, chapter ->
            chapter.apply { chapter_number = chapters.size.toFloat() - i }
        }
    }
```

Please test the code, as I wrote it directly on the GitHub review.

Reply:
Neat, first time using kotlin and I was thinking about doing this but couldn't quite figure out how. Your suggestions compiled and worked perfectly! 👌